frontend/src/component/ui/Toast.spec.tsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import { Toast } from './Toast';
+
+describe('Toast component', () => {
+  beforeAll(() => {
+    jest.useFakeTimers();
+  });
+
+  test('should render without crash', () => {
+    render(<Toast toastMessage={'Toast message.'} />);
+  });
+
+  test('should show content correctly when triggered to show', () => {
+    const toastRef = React.createRef<Toast>();
+    const { container } = render(
+      <Toast ref={toastRef} toastMessage={'Toast message.'} />
+    );
+
+    expect(container.textContent).not.toContain('Toast message.');
+    toastRef.current!.notify(1000);
+    expect(container.textContent).toContain('Toast message.');
+  });
+
+  test('should automatically hide content after delay', () => {
+    const toastRef = React.createRef<Toast>();
+    const { container } = render(
+      <Toast ref={toastRef} toastMessage={'Toast message.'} />
+    );
+
+    expect(container.textContent).not.toContain('Toast message.');
+    toastRef.current!.notify(2000);
+
+    jest.advanceTimersByTime(1000);
+    expect(container.textContent).toContain('Toast message.');
+
+    jest.advanceTimersByTime(1000);
+    expect(container.textContent).not.toContain('Toast message.');
+
+    jest.clearAllTimers();
+  });
+
+  test('second notify call should replace first toast', () => {
+    const toastRef = React.createRef<Toast>();
+    const { container } = render(
+      <Toast ref={toastRef} toastMessage={'Toast message.'} />
+    );
+
+    expect(container.textContent).not.toContain('Toast message.');
+    toastRef.current!.notify(2000);
+
+    jest.advanceTimersByTime(1000);
+    // second notify before the first one closes(at 1000ms)
+    toastRef.current!.notify(2000);
+    expect(container.textContent).toContain('Toast message.');
+
+    jest.advanceTimersByTime(1500);
+    // time is currently 2500ms
+    expect(container.textContent).toContain('Toast message.');
+
+    jest.advanceTimersByTime(1000);
+    // time is currently 3500ms
+    expect(container.textContent).not.toContain('Toast message.');
+
+    jest.clearAllTimers();
```

Comment:
```suggestion
    const toastRef = React.createRef<Toast>();
    let toastMessage = 'Toast Message';
    const { container } = render(
      <Toast ref={toastRef} toastMessage={toastMessage} />
    );

    expect(container.textContent).not.toContain('Toast message');
    toastRef.current!.notify(2000);

    jest.advanceTimersByTime(1000);
    
    toastMessage = 'New message';
    // second notify before the first one closes(at 1000ms)
    toastRef.current!.notify(2000);
    expect(container.textContent).toContain('New message');

    jest.advanceTimersByTime(1500);
    // time is currently 2500ms
    expect(container.textContent).toContain('New message');

    jest.advanceTimersByTime(1000);
    // time is currently 3500ms
    expect(container.textContent).not.toContain('New message');

    jest.clearAllTimers();
```

Reply:
@byliuyang, I don't think this will work. The component is created before changing the toast message for the second time.

Reply:
@rohithbalaji123 Yes, doesn't seem to work. It seems that we need this: https://testing-library.com/docs/example-update-props